sorc/build_ufs.sh

```diff
@@ -25,17 +25,47 @@ while getopts ":da:v" option; do
   esac
 done
 
-cd "${cwd}/ufs_model.fd"
+cd "${cwd}/ufs_model.fd" # TODO: This is redundant; see below.
 
 MAKE_OPT="-DAPP=${APP} -D32BIT=ON -DCCPP_SUITES=${CCPP_SUITES}"
 [[ ${BUILD_TYPE:-"Release"} = "DEBUG" ]] && MAKE_OPT+=" -DDEBUG=ON"
 COMPILE_NR=0
 CLEAN_BEFORE=YES
 CLEAN_AFTER=NO
 
-./tests/compile.sh "${MACHINE_ID}" "${MAKE_OPT}" "${COMPILE_NR}" "${CLEAN_BEFORE}" "${CLEAN_AFTER}"
-mv "./tests/fv3_${COMPILE_NR}.exe" ./tests/ufs_model.x
-mv "./tests/modules.fv3_${COMPILE_NR}.lua" ./tests/modules.ufs_model.lua
-cp "./modulefiles/ufs_common.lua" ./tests/ufs_common.lua
+if [[ "${MACHINE_ID}" != "noaacloud" ]]; then
+  ./tests/compile.sh "${MACHINE_ID}" "${MAKE_OPT}" "${COMPILE_NR}" "${CLEAN_BEFORE}" "${CLEAN_AFTER}"
+  mv "./tests/fv3_${COMPILE_NR}.exe" ./tests/ufs_model.x
+  mv "./tests/modules.fv3_${COMPILE_NR}.lua" ./tests/modules.ufs_model.lua
+  cp "./modulefiles/ufs_common.lua" ./tests/ufs_common.lua
+fi
+
+if [[ "${MACHINE_ID}" == "noaacloud" ]]; then
+  case $(dnsdomainname -f) in   
+     # TODO: Add Google and Azure platforms.
+      *pw-noaa*pw.local) CLOUD_MACHINE_ID="aws.${RT_COMPILER}" ;;
+      *) ;; 
+  esac
```

Comment:
we shouldn't be doing detection again.  MACHINE ID has been detected once. UFSWM is using a `PW_CSP` variable. Lets use that and if necessary use `PW_CSP` to distinguish.

Reply:
@aerorahul Can you point me to a use case within the UFSWM of the `PW_CSP` instance?

For clarification, the above detection was needed to determine which CSP the user is on. The DNS strings should be unique. This allows the GW (and build system) to determine which host they are on and thus make the downstream decisions accordingly.

For example, AWS PW use `mpiexec` as it's parallel executable launcher. However, Azure or Google could use another. This may ultimately wind up being redundant but since I haven't see the `PW_CSP` use case I felt it unsafe to make such an assumption.

Reply:
See https://github.com/ufs-community/ufs-weather-model/blob/develop/tests/detect_machine.sh
I would recommend a `PW_CSP` as a separate configurable if you wish to identify a CSP available on NOAACloud.